Waypoint doc updates (#15051)
istio/istio.io: content/en/docs/ambient/usage/waypoint/index.md

---

A **waypoint proxy** is an optional deployment of the Envoy-based proxy to add Layer 7 (L7) processing to a defined set of workloads.

Waypoint proxies are installed, upgraded and scaled independently from applications; an application owner should be unaware of their existence. Compared to the sidecar {{< gloss >}}data plane{{< /gloss >}} mode, which runs an instance of the Envoy proxy alongside each workload, the number of proxies required can be substantially reduced.

A waypoint, or set of waypoints, can be shared between applications with a similar security boundary. This might be all the instances of a particular workload, or all the workloads in a namespace.

> As opposed to {{< gloss >}}sidecar{{< /gloss >}} mode, in ambient mode policies are enforced by the **destination** waypoint. In many ways, the waypoint acts as a gateway into the namespace. Istio enforces that all traffic coming into the namespace goes through the waypoint, which then enforces all policies for that namespace.

Author: Would we then be best to change the next line about "coming into the namespace" etc, as it's not just about a namespace, it can also be for a service/workload?

> As opposed to {{< gloss >}}sidecar{{< /gloss >}} mode, in ambient mode policies are enforced by the **destination** waypoint. In many ways, the waypoint acts as a gateway into the namespace. Istio enforces that all traffic coming into the namespace goes through the waypoint, which then enforces all policies for that namespace.

Reviewer: I don't think I see that line but yeah, basically anywhere we imply that a waypoint is like ingress for a namespace I'd vote to reword that.

## Do you need a waypoint proxy?

The layered approach of ambient allows users to adopt Istio in a more incremental fashion, smoothly transitioning from no mesh, to the secure L4 overlay, to full L7 processing.

Most of the features of ambient mode are provided by the ztunnel node agent. Ztunnel is scoped to only process traffic at Layer 4 (L4), so that it can safely operate as a shared component.

> When you add a waypoint proxy for a workload, traffic will be forwarded from the ztunnel to that waypoint. If your applications require any of the following L7 mesh functions, you will need to use a waypoint proxy:

Author: Taking the suggestion for the second half.  Was the first half ok?  Are you worried about the fact that it's not just about adding the waypoint, but the waypoint has to be the right traffic type?

> When you add a waypoint proxy for a workload, traffic will be forwarded from the ztunnel to that waypoint. If your applications require any of the following L7 mesh functions, you will need to use a waypoint proxy:

Reviewer: maybe it's just me getting too caught up in the details but waypoint-for will matter as well as how the traffic was addressed. I am working on a doc around this now though so maybe we can just move on and caveat/link to my stuff once it is ready

> When you add a waypoint proxy for a workload, traffic will be forwarded from the ztunnel to that waypoint. If your applications require any of the following L7 mesh functions, you will need to use a waypoint proxy:

Reviewer: well, adding to a workload would, technically speaking, require an appropriate waypoint for as well as the client to be addressing traffic to that workload instead of a service. maybe something that moves away from specifics is the call

```suggestion
When you configure redirection to a waypoint, traffic will be forwarded by ztunnel to that waypoint. If your applications require any of the following L7 mesh functions, you will need to use a waypoint proxy:
```

* **Traffic management**: HTTP routing & load balancing, circuit breaking, rate limiting, fault injection, retries, timeouts
* **Security**: Rich authorization policies based on L7 primitives such as request type or HTTP header
* **Observability**: HTTP metrics, access logging, tracing

## Deploy a waypoint proxy

> Waypoint proxies are deployed declaratively using Kubernetes Gateway resources. The istioctl command can be used to generate a manifest for a waypoint.

Reviewer: ```suggestion
Waypoint proxies are deployed declaratively using Kubernetes Gateway resources. You can use istioctl experimental subcommands to generate, apply, and list these resoures.
```

namespace/default labeled
{{< /text >}}

> After a namespace is enrolled to use a waypoint, any requests from any pods enrolled in the mesh to any service running in that namespace will be routed through the waypoint for L7 processing and policy enforcement.

Reviewer: This is kind of misleading. Sidecar-enabled pods are part of the mesh but they won't use the waypoint...

Author: We stopped saying "in the ambient mesh" but this might be a reason to bring it back.
In a world where we have sidecars and waypoints, you would then need two policies - one with selectors and one with a `targetRef`?

Reviewer: I retract my complaint. Not due to being invalid but I was simply way too in the weeds on a detail we don't support for 1.22 any way. Having said that there is potentially a weird caveat to consider for ingress gateways too. I suppose "pods enrolled in ztunnel" might help imply that ingress gateways are their own special case 🤷‍♂️
